src/Application/Lockers/Commands/AddLocker/AddLockerCommand.cs

```diff
@@ -0,0 +1,62 @@
+﻿using Application.Common.Interfaces;
+using Application.Common.Models.Dtos.Physical;
+using AutoMapper;
+using Domain.Entities.Physical;
+using Domain.Exceptions;
+using MediatR;
+using Microsoft.EntityFrameworkCore;
+
+namespace Application.Lockers.Commands.AddLocker;
+
+public record AddLockerCommand : IRequest<LockerDto>
+{
+    public string Name { get; init; }
+    public string Description { get; init; }
+    public string RoomId { get; init; }
+    public int Capacity { get; init; }
+}
+
+public class AddLockerCommandHandler : IRequestHandler<AddLockerCommand, LockerDto>
+{
+    private readonly IApplicationDbContext _context;
+    private readonly IMapper _mapper;
+
+    public AddLockerCommandHandler(IApplicationDbContext context, IMapper mapper)
+    {
+        _context = context;
+        _mapper = mapper;
+    }
+
+    public async Task<LockerDto> Handle(AddLockerCommand request, CancellationToken cancellationToken)
+    {
+        var roomGuid = Guid.Parse(request.RoomId);
+        var room = await _context.Rooms.FirstOrDefaultAsync(x => x.Id == roomGuid, cancellationToken);
+
+        if (room is null)
+        {
+            throw new KeyNotFoundException("Room does not exist.");
+        }
+
+        if (room.NumberOfLockers == room.Capacity)
+        {
+            throw new LimitExceededException(
+                "This room cannot accept more folders. Please remove a folder then try again."
```

Comment:
this much elaboration is not required, this level of details is not needed in the api level
The first sentence is sufficient already, they don't need to know to "try again"
And also it is lockers, not folders

Reply:
ok, will fix